src/flextopo.jl

```diff
@@ -0,0 +1,795 @@
+    # Percolation flux from the root-zone to the slow storage sum classes [mm Δt⁻¹]
+    qpercolation_m::Vector{T} | "mm Δt-1"
+    # Evaporation from the root-zone storage, interception and hortonian [mm Δt⁻¹]
+    actevap::Vector{SVector{N,T}} | "mm Δt-1"
+    # Evaporation from the root-zone storage, interception and hortonian sum classes [mm Δt⁻¹]
+    actevap_m::Vector{T} | "mm Δt-1"
+    # Evaporation from the root-zone storage sum classes [mm Δt⁻¹]
+    rootevap_m::Vector{T} | "mm Δt-1"
+    #FAST 
+    # runoff from fast reservoir [mm Δt⁻¹]
+    qfast::Vector{SVector{N,T}} | "mm Δt-1"
+    #SLOW
+    # runoff from slow reservoir [mm Δt⁻¹]
+    qslow::Vector{T} | "mm Δt-1"
+    #Total [mm Δt⁻¹]
+    runoff::Vector{T} | "mm Δt-1"
+    # fast runoff sum classes [mm Δt⁻¹]
+    qfast_tot = Vector{T} | "mm Δt-1"
+
+
+    ## WATERBALANCES
+    #water balance snow store
+    wb_snow::Vector{T} | "mm Δt-1"
+    #water balance interception storage [mm Δt⁻¹]
+    wb_interception::Vector{SVector{N,T}} | "mm Δt-1"
+    #water balance hortonian ponding storage [mm Δt⁻¹]
+    wb_hortonponding::Vector{SVector{N,T}} | "mm Δt-1"
+    #water balance hortonian runoff storage [mm Δt⁻¹]
+    wb_hortonrunoff::Vector{SVector{N,T}} | "mm Δt-1"
+    #water balance root-zone storage [mm Δt⁻¹]
+    wb_rootzone::Vector{SVector{N,T}} | "mm Δt-1"
+    #water balance fast storage [mm Δt⁻¹]
+    wb_fast::Vector{SVector{N,T}} | "mm Δt-1"
+    #water balance slow storage [mm Δt⁻¹]
+    wb_slow::Vector{T} | "mm Δt-1"
+    #total water balance [mm Δt⁻¹]
+    wb_tot::Vector{T} | "mm Δt-1"
+
+
+    # function FLEXTOPO{T,N}(args...) where {T,N}
+    #     equal_size_vectors(args)
+    #     return new(args...)
+    # end
+end
+
+statevars(::FLEXTOPO) = (:snow, :snowwater, :interceptionstorage, :hortonpondingstorage, :hortonrunoffstorage, :rootzonestorage, :faststorage, :slowstorage)
+
+function common_snow_hbv(flextopo::FLEXTOPO, config)
+    for i = 1:flextopo.n
+        # precip correction 
+        precipcorr = flextopo.precipitation[i] * flextopo.pcorr[i]
+
+        #hbv snow 
+        snow, snowwater, snowmelt, rainfallplusmelt, snowfall = snowpack_hbv(
+            flextopo.snow[i],
+            flextopo.snowwater[i],
+            precipcorr,
+            flextopo.temperature[i],
+            flextopo.tti[i],
+            flextopo.tt[i],
+            flextopo.ttm[i],
+            flextopo.cfmax[i],
+            flextopo.whc[i],
+        )
+
+        #update stores
+        # states_ is sum of states of previous time steps to compute WB at the end (per class); states_m is for combined stores
+        flextopo.states_m[i] = flextopo.snow[i] + flextopo.snowwater[i]
+        flextopo.snowfall[i] = snowfall
+        flextopo.snowmelt[i] = snowmelt
+        flextopo.snow[i] = snow
+        flextopo.snowwater[i] = snowwater
+        flextopo.rainfallplusmelt[i] = rainfallplusmelt
+        flextopo.precipcorr[i] = precipcorr
+        #TODO add wb_snow wbSww
```

Comment:
Does this need to be addressed before merging this PR?

Reply:
good point, this is now added in 7dde6ff6469d2ec5e4b975f9dd0f455987dff313